### PR TITLE
Add block size default

### PR DESCRIPTION
**Pull request by kstribrnAmzn**
Description
-----------
Block size for file transfer
should be configurable from 256 bytes to
128 KB.

Test Steps
-----------
To be completed

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
https://github.com/aws/aws-iot-core-mqtt-file-streams-embedded-c/issues/24

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/memory_statistics_config.json [hunk 1]
```diff
@@ -13,5 +13,8 @@
         "source/include",
         "coreJSON/source/include",
         "tinycbor/src"
+    ],
+    "compiler_flags": [
+        "MQTT_STREAMS_DO_NOT_USE_CUSTOM_CONFIG"
     ]
 }
```
Comment (paulbartell) [inline .github/memory_statistics_config.json:20]: /nit no blank line at the end of file.

File: test/unit-test/MQTTFileDownloader_config.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * @file MQTTFileDownloader_config.h
+ * @brief Testing values for various MQTT stream macros.
+ */
+
+#ifndef MQTT_FILE_DOWNLOADER_CONFIG_H
+#define MQTT_FILE_DOWNLOADER_CONFIG_H
+
+#include <stdint.h>
+
+#define mqttFileDownloader_CONFIG_BLOCK_SIZE    256U
```
Comment (paulbartell) [inline test/unit-test/MQTTFileDownloader_config.h:21]: Perhaps the default should be specified so that the payload is 4k

**Review by paulbartell — COMMENTED** [review]
A couple of small suggestions

File: .github/memory_statistics_config.json (hunk 1 above)
Comment (kstribrnAmzn) [reply .github/memory_statistics_config.json:20]: I'll add the newline

File: test/unit-test/MQTTFileDownloader_config.h (hunk 2 above)
Comment (kstribrnAmzn) [reply test/unit-test/MQTTFileDownloader_config.h:21]: I'm fine with this. That would match the old OTA library better.

**Review by paulbartell — APPROVED** [review]

**Review by rawalexe — APPROVED** [review]

**Pull request merged**